fireshape/innerproduct.py

```diff
@@ -245,21 +245,38 @@ def get_nullspace(self, V):
         """
         This nullspace contains constant functions as well as rotations.
         """
+        # Elasticity null space is different for periodic domains.
+        # This code assumes that the domain is periodic on every bdry.
+        # If the domain is only partially periodic, modify the nullspace or
+        # specify at least 1 DirichletBC to empty the nullspace
+        element = self.Q.mesh_r.coordinates.function_space().ufl_element()
+        is_periodic = False
+        if element.family() == 'Discontinuous Lagrange':
```

Comment:
isn't it better to use `Q.is_DG` ?

Reply:
That was my original thought as well. However it broke a bunch of tests because other ControlSpaces don't have that option. It didn't make sense say to extend it to the other control spaces.

Reply:
solved